src/network/core.rs

```diff
@@ -232,6 +232,210 @@ impl Core {
         }
     }
 
+    pub fn macvlan_per_podman_network(
+        per_network_opts: &types::PerNetworkOptions,
+        network: &types::Network,
+        netns: &str,
+    ) -> Result<types::StatusBlock, std::io::Error> {
+        //  StatusBlock response
+        //  StatusBlock response
+        let mut response = types::StatusBlock {
+            interfaces: Some(HashMap::new()),
+        };
+        // Default MACVLAN_MODE to bridge or get from driver options
+        let mut macvlan_mode: u32 = constants::MACVLAN_MODE_BRIDGE;
+        if let Some(options_map) = network.options.as_ref() {
+            if let Some(mode) = options_map.get("mode") {
+                match core_utils::CoreUtils::get_macvlan_mode_from_string(mode) {
+                    Ok(mode) => {
+                        macvlan_mode = mode;
+                    }
+                    Err(err) => {
+                        return Err(err);
+                    }
+                }
+            }
+        }
+
+        // get master interface name
+        let mut master_ifname: String = match network.network_interface.as_ref() {
+            None => {
+                if let Ok(ifname) = core_utils::CoreUtils::get_default_route_interface() {
+                    ifname
+                } else {
+                    return Err(std::io::Error::new(
+                        std::io::ErrorKind::Other,
+                        "unable to find any valid master interface for macvlan".to_string(),
+                    ));
+                }
+            }
+            Some(interface) => interface.to_owned(),
+        };
+
+        // user could provide empty value as well so handle that
+        if master_ifname.is_empty() {
+            if let Ok(ifname) = core_utils::CoreUtils::get_default_route_interface() {
```

Comment:
Why an if-let here? It seems like we should be returning an error if we can't get a default interface name, because ifname will still be empty at the end of this block and we'll just proceed on and fail later.

Reply:
@mheon Yes that is handled below in  `else` block if result is not `Ok` and another block below handles the case for empty check.

Reply:
What else block? I don't see one

Reply:
@mheon added a `else` my bad got confused with the one above.

Reply:
https://github.com/containers/netavark/pull/76/files#diff-dadc3c975feb7c242a4ea465a9dcab3a98600a15ce4c5632ad5a671c0e3d209aR285